test/e2e-go/features/stateproofs/stateproofs_test.go

```diff
@@ -173,3 +175,154 @@ func verifyStateProofForRound(r *require.Assertions, libgoal libgoal.Client, res
 	r.NoError(err)
 	return stateProofMessage, nextStateProofBlock
 }
+
+func TestStateProofsRecovery(t *testing.T) {
```

Comment:
It is not clear what this test is about. Some comments will be appreciated here.